src/assets/scripts/client/measurement/MeasureTool.js

```diff
@@ -0,0 +1,363 @@
+        const initialTurn = this._getInitialTurnParameters();
+        const turnRadius = initialTurn !== null ? initialTurn.turnRadius : 0;
+        const pointsList = [...this._points]; // Shallow copy as the first point may be replaced
+
+        // These are the values used when reducing the points array
+        const initialValues = {
+            previousLeg: null,
+            totalDistance: 0,
+            totalDuration: 0
+        };
+
+        // If there is an initialTurn (eg. a turn onto the first leg), then the exit point is used
+        // as the first point in the path
+        // We also calculate the distance and time take to fly the turn
+        if (initialTurn !== null) {
+            const { exitPoint, arcLength } = initialTurn;
+            const distance = nm(arcLength);
+            const duration = Math.round((distance / groundSpeed) * TIME.ONE_HOUR_IN_SECONDS);
+
+            initialValues.lastPoint = exitPoint;
+            initialValues.totalDistance = distance;
+            initialValues.totalDuration = duration;
+
+            pointsList[0] = exitPoint;
+        }
+
+        // The first leg is just an end point
+        const firstPoint = this._getRelativePosition(pointsList.shift());
+        const firstLeg = new MeasureLegModel(firstPoint);
+        initialValues.previousLeg = firstLeg;
+
+        pointsList.reduce((lastValues, point) => {
+            const { previousLeg } = lastValues;
+            let { totalDistance, totalDuration } = lastValues;
+            const leg = new MeasureLegModel(
+                this._getRelativePosition(point),
+                turnRadius,
+                previousLeg
+            );
+            const distance = nm(leg.distance);
+            const bearing = heading_to_string(leg.bearing);
+            let duration = 0;
+
+            if (groundSpeed !== null) {
+                duration = Math.round((distance / groundSpeed) * TIME.ONE_HOUR_IN_SECONDS);
+            }
+
+            const labels = [
+                this._getLabel(distance, duration, bearing)
+            ];
+
+            totalDistance += distance;
+            totalDuration += duration;
+
+            if (totalDistance !== distance) {
+                labels.push(this._getLabel(totalDistance, totalDuration));
+            }
+
+            leg.labels = labels;
+
+            return {
+                previousLeg: leg,
+                totalDistance,
+                totalDuration
+            };
+        }, initialValues);
+
+        return {
+            initialTurn,
+            firstLeg
+        };
+    }
+
+    /**
+     * Removes the last point from the path
+     *
+     * @for MeasureTool
+     * @method removeLastPoint
+     */
+    removeLastPoint() {
+        // The "last" point is actually the 2nd last item in _points
+        // This means the path will end at the cursor position when redrawn
+        const { length } = this._points;
+        if (length > 2) {
```

Comment:
missing empty line above `if`

Reply:
I don't know how you spot these, I always think I've done it.